File: Leetcode/isIsomorphic.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character, Character> map = new HashMap<Character, Character>();
        HashMap<Character, Boolean> mapChar = new HashMap<Character, Boolean>();
        int size = s.length();
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < size; i++) {
            // klo ada key nya tinggal ganti sm key ny
            if(map.containsKey(s.charAt(i))) {
                sb.append(map.get(s.charAt(i))); 
            } else {
                // klo gaada ganti jg trs mapping
                if(mapChar.containsKey(t.charAt(i)) && !map.containsKey(s.charAt(i))){
                    sb.append("-");
                    continue;
                }
                map.put(s.charAt(i), t.charAt(i));
                mapChar.put(t.charAt(i), true);
                sb.append(t.charAt(i));
            }
        }
        return sb.toString().equals(t);
    }
}

File: Leetcode/isIsomorphic.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        int sSize = s.size();
        int tSize = t.size();
        if(sSize != tSize) return false;
        
        map<char, char> m;
        map<char, bool> seen;
        for(int i = 0; i < sSize; i++){
            if(seen.find(t[i]) == seen.end()){
                m[s[i]] = t[i];
                seen[t[i]] = true;
            } else if(m.find(s[i]) != m.end()) continue;
            else m[s[i]] = '!';
        }

        for(int i = 0; i < sSize; i++){
            if(t[i] != m[s[i]]) return false;
        }

        return true;
    }
};
